pkg/prometheus/statefulset.go

```diff
@@ -381,6 +381,10 @@ func makeStatefulSetSpec(p monitoringv1.Prometheus, c *Config, ruleConfigMapName
 		securityContext = p.Spec.SecurityContext
 	}
 
+  if p.Spec.EnableAdminAPI == "true" {
```

Comment:
as this is a bool, it would just be

```suggestion
  if p.Spec.EnableAdminAPI {
```

it also needs to be formatted correctly